mkosi/__init__.py

```diff
@@ -6932,6 +6932,12 @@ def setup_ssh(
                               ListenStream=
                               ListenStream={args.ssh_port}
                               """)
+
+        add_dropin_config(root, "ssh@.service", "runtime-directory-preserve",
```

Comment:
Great work debugging this! 

I'm a bit surprised seeing the template unit here, is that because of the `ssh.socket` unit we use? Socket activation for SSH is a bit of a niche thing (Arch stopped defaulting to that because of some potential DoS issue) and on a sufficiently new OpenSSH (8.4 in Debian bullseye seems to be enough) this could also be done with a config file in `/etc/ssh/sshd_config.d`, which are pulled in by the snippet at the top of
```
# /etc/ssh/sshd_config
Include /etc/ssh/sshd_config.d/*.conf
```
shipped by default in Debian.

Reply:
I enabled this - it seemed much nicer to use socket activation

Reply:
Ah, now I remember! :)

Looking at the Arch wiki, [this is the bug report](https://bugs.archlinux.org/task/62248) about the DoS issue. Apparently upstream used to ship the socket and template service units, but stopped with OpenSSH 8.0 due to the trigger limit. 

I very much like socket activation, but maybe that's something to go with the upstream default, which would also bring Debian in line with the rest. This is nothing, though, that this PR needs to do.

Reply:
@bluca You introduced usage of the socket activation for SSH when you added the ssh port option. I'm wondering if we shouldn't stick to the default usage of sshd instead. I also realized that it doesn't seem like we support usage of --ssh-port unless the socket unit on Debian/Ubuntu is used.

Reply:
As long as the port can be changed easily, that's fine - it seemed much more robust to use the socket unit, since support for drop-ins sshd config snippets is relatively new and not guaranteed to work on older ubuntu/debian releases, while the socket always works